### PR TITLE
feat(EMS-921-1217): Application submission 

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds the application submission functionality.

# Changes
- Setup new GQL resolver, `submitApplication`. This function changes the application status to submitted, adds a submission date and (new field) `previousStatus` for future iterations/integrations.
- Update the declarations "how your data will be used" POST controller to call the submit application GQL mutation.
- Setup route, controller and page for the "application submitted" page. Ensure that if an application does not have a submitted status, we redirect to the "all sections" page.
- Create a new cypress command for the full journey of creating/signing into an account and then completing and submitting an application: `completeSignInAndSubmitAnApplication`
- E2E test coverage

### CODE REVIEW

File: e2e-tests/cypress/support/insurance/complete-sign-in-and-submit-an-application.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import completeSignInAndGoToApplication from './account/complete-sign-in-and-go-to-application';
+
+/**
+ * completeSignInAndSubmitAnApplication
```
Comment (abhi-markan) [inline e2e-tests/cypress/support/insurance/complete-sign-in-and-submit-an-application.js:4]: One line function explanation please.

File: src/api/custom-resolvers/submit-application.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Context } from '.keystone/types'; // eslint-disable-line
+import { APPLICATION } from '../constants';
+import { SubmitApplicationVariables, SubmitApplicationResponse } from '../types';
+
+/**
+ * submitApplication
```
Comment (abhi-markan) [inline src/api/custom-resolvers/submit-application.ts:6]: Same as above please.

File: src/api/custom-resolvers/submit-application.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          where: { id: application.id },
+          data: update,
+        });
+
+        return {
+          success: true,
+        };
+      }
+
+      console.info('Unable to submit application - application already submitted');
```
Comment (abhi-markan) [inline src/api/custom-resolvers/submit-application.ts:46]: Should this be `console.error`?

File: src/api/custom-resolvers/submit-application.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    console.info('Unable to submit application - no application found');
```
Comment (abhi-markan) [inline src/api/custom-resolvers/submit-application.ts:49]: Should this be `console.error`?

File: src/ui/styles/styles.scss [hunk 5]
```diff
@@ -43,3 +43,7 @@ $govuk-new-link-styles: true;
   display: -webkit-flex;
   display: flex;
 }
+
+.ukef-application-submitted {
+  max-width: 360px;
```
Comment (abhi-markan) [inline src/ui/styles/styles.scss:48]: Where has this value been derived from please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Overall looks good, few comments to address please.

File: src/api/custom-resolvers/submit-application.ts (hunk 3 above)
Comment (ttbarnes) [reply src/api/custom-resolvers/submit-application.ts]: Good shout, updated

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**